src/scripts/main.js

```diff
@@ -2,4 +2,27 @@
 
 document.addEventListener('click', (e) => {
   // write code here
+  const spider = document.querySelector('.spider');
+  const wall = e.target.closest('.wall');
+
+  if (!wall) {
+    return;
+  }
+
+  const wallRect = wall.getBoundingClientRect();
+  const borderSize = parseInt(getComputedStyle(wall).borderWidth);
+  const spiderRect = spider.getBoundingClientRect();
+
+  let targetX = e.clientX - wallRect.left - spiderRect.width / 2 - borderSize;
+  let targetY = e.clientY - wallRect.top - spiderRect.height / 2 - borderSize;
+
+  targetX = Math.max(0, Math.min(targetX, wall.clientWidth - spiderRect.width));
+
+  targetY = Math.max(
+    0,
+    Math.min(targetY, wall.clientHeight - spiderRect.height),
+  );
+
+  spider.style.left = `${targetX}px`;
```

Comment:
This line is changing the style of the spider directly. This can lead to difficulties in managing styles later. Consider adding a CSS class to the spider instead, and defining the style in your CSS file.